File: List/InsertionSortList/InsertSortList.java
/*
class ListNode {
	int val;
	ListNode next;
	
	ListNode(int val) {
		this.val = val;
		this.next = null;
	}
};
*/
class Solution {
	public	ListNode insertSortList(ListNode head) {
		if(head == null)
			return null;
		
		ListNode dummy = new ListNode(0);
		ListNode node = head;
		while(curr != null) {
			ListNode prev = dummy;
			while(prev.next != null && prev.next.val < curr.val) {
				prev = prev.next;
			}
			ListNode tmp = curr.next;
			curr.next = prev.next;
			prev.next = curr;
			curr = tmp;
			
		}
		return dummy.next;
	}
}

File: List/InsertionSortList/InsertSortList.cc
/*
class ListNode {
	int val;
	ListNode* next;
	
	ListNode(int v):val(v), next(null) {}
};
*/

class Solution {
	public:
		ListNode* insertSortList(ListNode* head) {
			if(head == NULL)
				return NULL;
			
			ListNode *dummy = new ListNode(0);
			ListNode *node = head;
			while(curr != NULL) {
				ListNode prev = dummy;
				while(prev->next != NULL && prev->next->val < curr->val) {
					prev = prev->next;
				}
				ListNode tmp = curr->next;
				curr->next = prev->next;
				prev->next = curr;
				curr = tmp;
				
			}
			return dummy->next;
		}
};